### PR TITLE
Upgrade to flow@0.37 / fix some typings

### PR DESCRIPTION
**Pull request by ryyppy**
* Adds inquirer libdefs for more exact typings
* Refine / Fix some flowtype errors (mostly in `__tests__`)
* Fix a small bug in `src/util/map.js` found by flow@0.37

**Summary**

This PR adds flow-typed definitions for `inquirer`. Nice thing is, I didn't even need to touch the code after adding the definitions (I played around with it and could produce errors when passing wrong parameters to `inquirer.prompt()`).

Also I invested some time to upgrade `flow` from `0.33` to the most recent version `0.37.3`, fixing some typing issues / bugs along the way.

**Test plan**

When I originally pulled the source from `master`, I realized that 11 tests were failing.
So I was doing two test runs, one testing the HEAD and one testing my modifications.

The same number of tests failed.
Also I made sure that `flow@0.37.3` type checks the source properly.

### CODE REVIEW

**Comment by ryyppy** [comment]
@bestander What is happening with CI right now? Any open issue tackling this problem? Seems that several tests are failing due to some changes

**Comment by bestander** [comment]
@ryyppy, I fixed this, could you rebase please?

**Comment by ryyppy** [comment]
Hmm, not sure what's happening here:

image

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "defaults": "^1.0.3",
     "detect-indent": "^4.0.0",
     "diff": "^2.2.1",
+    "flow-bin": "^0.37.3",
```
Comment (bestander) [inline package.json]: That is the reason why Circle failed.
I'll send a quick fix

**Comment by ryyppy** [comment]
Awesome, thanks 👍

**Pull request merged**